### PR TITLE
stop badge flashing when looking at a convo w/ incoming msgs

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers

### CODE REVIEW

**Comment by chrisnojima** [comment]
if you're in a convo and you setup a terminal to spam you w/ messages you'll see the badge doens't go from 1/0/1/0 quickly like it does on master

File: shared/native/notification-listeners.shared.js [hunk 1]
```diff
@@ -10,9 +11,19 @@ import type {incomingCallMapType} from '../constants/types/flow-types'
...
 
 export default function (dispatch: Dispatch, getState: () => Object, notify: any): incomingCallMapType {
+  const throttledDispatch = throttle(action => dispatch(action), 1000, {leading: false, trailing: true})
   return {
     'keybase.1.NotifyBadges.badgeState': ({badgeState}) => {
-      dispatch(receivedBadgeState(badgeState))
+      const totalChats = (badgeState.conversations || []).reduce((total, c) => total + c.UnreadMessages, 0)
+
+      const action = receivedBadgeState(badgeState)
+      if (totalChats) {
```
Comment (gabriel) [inline shared/native/notification-listeners.shared.js]: `if (totalChats > 0)` is more understandable?

**Comment by cjb** [comment]
Just curious -- since it's a little undesirable to have new messages show up 1s before the badge does, did you do any measuring of how long it takes to go from a badge of 1 to a badge of 0, to try to optimize that number a bit for most people?

File: shared/native/notification-listeners.shared.js (hunk 1 above)
Comment (chrisnojima) [inline shared/native/notification-listeners.shared.js]: sure

**Comment by chrisnojima** [comment]
@cjb I did not. Just keeping things simple and didn't want to make any assumptions. If we feel like this is a really big problem and hurting us then we can do something more complex. I think with badging there isn't an expectation of it being realtime on most platforms so having a lag of 1 second seems fine (you'll also get system notifications and stuff like that as well)

**Pull request merged**